File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public static int lengthOfLongestSubstring(String s) {
        int count = 0;
        int maxCount = 0;
        String tempStr = new String();
        for(int i = 0; i < s.length(); i++){
            count = 0;
            for(int j = i; j < s.length(); j++){
                if(!tempStr.contains(s.substring( j, j+1))){
                    count++;
                    tempStr += s.substring(j, j+1);
                    if(maxCount < count){
                        maxCount = count;
                    }
                
                }else{
                    tempStr = new String();
                    break;
                }
            }
        }
        return maxCount;
    }

}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        start = 0;
        end = 0;
        _max = 0;
        _set = set()
        while(end < len(s)):
            if(s[end] not in _set):
                _set.add(s[end])
                end += 1
                _max = max(_max, len(_set))
            else:
                _set.remove(s[start])
                start += 1
        
        return _max